packages/create-webstone-app/src/index.ts

```diff
@@ -1,6 +1,155 @@
+  create(cwd, {
+    name: appName,
+    template: type === "app" ? "skeleton" : "skeletonlib",
+    types: "typescript",
+    eslint: true,
+    prettier: true,
+    playwright: true,
+    vitest: true,
+  });
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function sortKeys(obj: any) {
+  if (typeof obj !== "object" || obj === null)
+    throw new Error("Invalid object");
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const sortedObj: Record<string, any> = {};
+  Object.keys(obj)
+    .sort()
+    .forEach((key) => {
+      sortedObj[key] = obj[key];
+    });
+  return sortedObj;
+}
+
+function updatePackageJSON(
+  cwd: string,
+  options: { type: WebstoneAppType; extendCLI: boolean },
+) {
+  const { type, extendCLI } = options;
+  const pkg: PackageJson = fs.readJSONSync(path.resolve(`${cwd}/package.json`));
+  let newPkg: PackageJson = pkg;
+  if (type === "app") {
+    newPkg = deepMergeWithSortedKeys(pkg, appPackageJson);
+  }
+  if (type === "plugin" && extendCLI) {
+    newPkg = deepMergeWithSortedKeys(pkg, pluginPackageJson);
+  }
+
+  fs.writeJsonSync(path.resolve(`${cwd}/package.json`), newPkg, {
+    encoding: "utf-8",
+    spaces: "\t",
+  });
+}
+
+function deepMergeWithSortedKeys(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  target: Record<string, any>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  source: Record<string, any>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Record<string, any> {
+  const merged = merge(target, source);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function recursiveSort(obj: Record<string, any>): Record<string, any> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const sortedObj: Record<string, any> = {};
+
+    Object.keys(obj).forEach((key) => {
+      if (["dependencies", "devdependencies", "scripts"].includes(key)) {
+        sortedObj[key] = sortKeys(obj[key]);
+      } else {
+        sortedObj[key] = obj[key];
+      }
+    });
+
+    return sortedObj;
+  }
+
+  return recursiveSort(merged);
+}
+
+function copyBuildScript(cwd: string) {
+  fs.copySync(
+    new URL("../templates/plugin-structure/build-cli.js", import.meta.url)
+      .pathname,
+    `${cwd}/scripts/build-cli.js`,
+  );
+}
+
+function copyCLIExtension(cwd: string) {
+  // copy command
+  fs.copySync(
+    new URL("../templates/plugin-structure/command.ts", import.meta.url)
+      .pathname,
+    `${cwd}/src/cli/commands/${getRawAppName(cwd)}/hello-world.ts`,
+  );
+
+  //copy extension
+  fs.copySync(
+    new URL("../templates/plugin-structure/extension.ts", import.meta.url)
+      .pathname,
+    `${cwd}/src/cli/extensions/hello-world.ts`,
+  );
+
+  // copy templates
+  fs.copySync(
+    new URL("../templates/plugin-structure/template.ejs", import.meta.url)
+      .pathname,
+    `${cwd}/src/cli/templates/template.ejs`,
+  );
+}
```

Comment:
That's maybe a bit ugly still, it's doing its job t hough

Reply:
Looks fine to me